### PR TITLE
feat: add Off cli

### LINKED ISSUE #132: `pop off` cli command
Gracefully shutdown the pop with a CLI command. `shutdown` might be a more appropriate name but `off` seems shorter and fun.

### PR DESCRIPTION
**Pull request by gallexis**
closes #132

### CODE REVIEW

File: node/popn.go [hunk 1]
```diff
@@ -306,6 +313,14 @@ func (nd *node) send(n Notify) {
 	}
 }
 
+// Off shutdown the node gracefully
+func (nd *node) Off(ctx context.Context) {
+	nd.send(Notify{OffResult: &OffResult{}})
+	fmt.Println("Gracefully shutdown node")
```
Comment (tchardin) [inline node/popn.go:319]:
```suggestion
	fmt.Println("==> Shut down pop daemon")
```

File: node/popn.go [hunk 2]
```diff
@@ -95,6 +95,8 @@ type Options struct {
 	Capacity uint64
 	// ReplInterval defines how often the node attempts to find new content from connected peers
 	ReplInterval time.Duration
+	// GracefulShutdown is the CancelFunc used for gracefully shutting down the node
+	GracefulShutdown context.CancelFunc
```
Comment (tchardin) [inline node/popn.go:99]:
```suggestion
	CancelFunc context.CancelFunc
```

File: cmd/pop/cli/off.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+			prc <- pr
+		}
+	})
+	go receive(ctx, cc, c)
+
+	cc.Off()
+
+	select {
+	case <-prc:
+		fmt.Println("Successfully sent Graceful Shutdown command")
```
Comment (tchardin) [inline cmd/pop/cli/off.go:34]:
```suggestion
		fmt.Println("pop daemon is shutting down")
```

File: cmd/pop/cli/off.go [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+	"fmt"
+	"github.com/myelnet/pop/node"
+	"github.com/peterbourgon/ff/v3/ffcli"
+)
+
+var offCmd = &ffcli.Command{
+	Name:       "off",
+	ShortUsage: "off",
+	ShortHelp:  "Gracefully shutdown the Pop daemon",
+	LongHelp:   "The 'pop off' command gracefully shutdown the Pop daemon.",
```
Comment (tchardin) [inline cmd/pop/cli/off.go:14]:
```suggestion
	LongHelp:   "The 'pop off' command gracefully shuts down the Pop daemon.",
```

**Review by tchardin — APPROVED** [review]
Just a few nits to keep the copy and variable naming consistent. Thanks

**Comment by gallexis** [comment]
Hum I need to fix my rebase

File: cmd/pop/cli/off.go [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+	"context"
+	"fmt"
+	"github.com/myelnet/pop/node"
+	"github.com/peterbourgon/ff/v3/ffcli"
+)
+
+var offCmd = &ffcli.Command{
+	Name:       "off",
+	ShortUsage: "off",
+	ShortHelp:  "Gracefully shutdown the Pop daemon",
```
Comment (gallexis) [inline cmd/pop/cli/off.go:13]: @tchardin shouldn' it be "Gracefully shuts down..." as well ?

Comment (tchardin) [reply cmd/pop/cli/off.go:13]: @gallexis 👍

**Pull request merged**